Cython/Compiler/Nodes.py

```diff
@@ -3899,14 +3954,21 @@ def generate_arg_assignment(self, arg, item, code):
     def generate_stararg_init_code(self, max_positional_args, code):
         if self.starstar_arg:
             self.starstar_arg.entry.xdecref_cleanup = 0
-            code.putln('%s = PyDict_New(); if (unlikely(!%s)) return %s;' % (
-                self.starstar_arg.entry.cname,
-                self.starstar_arg.entry.cname,
-                self.error_value()))
+
+            if not _get_type_attr(self.starstar_arg, "is_fastcall_dict"):
```

Comment:
Phew, code like this is really a big hack. There should be some general way to handle this. fastcall-dicts and -tuples really aren't special enough for this.